adobe_vipm/flows/sync.py

```diff
@@ -43,11 +45,118 @@
     notify_missing_prices,
 )
 from adobe_vipm.flows.utils.notification import notify_processing_lost_customer
+from adobe_vipm.notifications import send_exception
 from adobe_vipm.utils import get_3yc_commitment, get_commitment_start_date, get_partial_sku
 
 logger = logging.getLogger(__name__)
 
 
+def _add_missing_subscriptions(
+    mpt_client: MPTClient,
+    adobe_client: AdobeClient,
+    customer: dict,
+    agreement: dict,
+    subscriptions_for_update: tuple[str],
+) -> None:
+    logger.info(f"Checking missing subscriptions for {agreement["id"]=}")
+    adobe_subscriptions = adobe_client.get_subscriptions(
```

Comment:
I think the problem with this function is that it returns all the subscriptions for customer, including all the deployment subscriptions that belongs to another deployment agreements.